### DISCUSSION TITLE
How to add a simple serverless function?

### DISCUSSION THREAD

**Discussion by brettdewoody** [11ty/eleventy#3309] [Q&A]
I'd like to add a simple serverless function on an Eleventy site running on Netlify. As an example, I want to create a function to create a Stripe checkout session. 

Is the `EleventyServerlessBundlerPlugin` necessary for this?

Or can I create a function in `/netlify/functions/` and have this deployed as a Netlify function? 

And for local development, would this work with `npx eleventy`, or would it require running `netlify dev`?

**Comment by cfjedimaster** [comment]
Most likely you would want a Netlify function, and would use netlify dev to test your site (and it works just fine running Eleventy as well). Eleventy Serverless is more for cases where you need to run something on the server and make use of Eleventy layout and such. I'm probably grossly over simplifying that and it's been a while since I used it, but again, you probably just want the Netlify function.

**Reply by brettdewoody** [reply]
That's what I'm thinking as well.

I think the issue is I currently use `npx eleventy` for `dev`, combined with a few other watch commands (like `npx tailwindcss`), so I'm having trouble getting `netlify dev` to play nice.

**Reply by cfjedimaster** [reply]
Check the help for netlify dev, there's options for passing additional info and such. So for example, I do: `netlify dev -c "eleventy --serve --quiet"`
